src/utils/firebase.tsx

```diff
@@ -0,0 +1,33 @@
+// Import the functions you need from the SDKs you need
+import { initializeApp } from "firebase/app";
+import { getAuth, signInWithPopup, GithubAuthProvider } from "firebase/auth";
+import { firebaseConfig } from './firebaseConfig';
+
+// Initialize Firebase
+initializeApp(firebaseConfig);
+
+const provider = new GithubAuthProvider();
+
+const auth = getAuth();
+
+// Referenced https://firebase.google.com/docs/auth/web/github-auth
+export const signInWithGithub = async () => {
+    try {
+        const result = await signInWithPopup(auth, provider);
+        // you can access user's information with result.user
+        const credential = GithubAuthProvider.credentialFromResult(result);
+        const token = credential?.accessToken;
+
+        // send a request to Backend after signed up
+        fetch('http://localhost:5000/service/signup', {
+            headers: {
+                "Authorization": `Bearer ${token}`,
+            }
+        }).then(response => response.json());
+
+    } catch (err) {
+        // Handle errors
+        console.log(err.errorMessage);
+        console.log("ERROR", err);
+    }
+};
```

Comment:
I'm just a little confused as this is a typescript file. How does it work in relation to the rest of the flask server?

Reply:
The front-end needs to get the user's information, including its token after requesting the Github authentication. It will use this token to make any requests to the backend. When the backend receives this token, it will verify that this one is correct, and if it is, it will return the user's information. Basically, Client calls Github Authentication Firebase, user authenticates and a token is saved in Firebase, token is then sent to any requests to the server, serves verifies that this is a valid token found in Firebase, if it's valid, it uses that user information to handle the requests.

Reply:
Oh yeah makes sense. I thought this was in `server/` not `src/` and that's where the confusion came from.